internal/future/statestore/redis.go

```diff
@@ -73,7 +73,8 @@ func newRedis(cfg config.View) (Service, error) {
 	// always returns a valid connection, and will just fail on the first
 	// query: https://godoc.org/github.com/gomodule/redigo/redis#Pool.Get
 	redisConn := pool.Get()
-	defer redisCloser(redisConn.Close)
+	defer handleConnectionClose(&redisConn)
```

Comment:
Curious why this had to change from `redisCloser`? Is there really a problem here?

Reply:
Long story.. I did the lint fix from the bottom of the file to the top so I implement this handleConnectionClose method to reuse the code, and it was when I reached to the top of this file I saw and realized we have this redisCloser method already :)